frontend/model/state.js

```diff
@@ -524,78 +524,59 @@ const getters = {
         return nameA > nameB ? 1 : -1
       })
   },
-  ourPrivateDirectMessages (state, getters) {
-    const privateDMs = {}
-    const contractIDs = Object.keys(getters.ourDirectMessages)
-      .filter(cID => getters.ourDirectMessages[cID].privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE && state[cID])
-    for (const contractID of contractIDs) {
-      const usernames = Object.keys(state[contractID].users || {}) // NOTE: empty object is used while syncing the contract
-      const partner = usernames[0] === getters.ourUsername ? usernames[1] : usernames[0]
-      if (partner) {
-        privateDMs[partner] = {
-          ...getters.ourDirectMessages[contractID],
-          contractID
+  ourGroupDirectMessages (state, getters) {
+    const currentGroupDirectMessages = {}
+    for (const chatRoomId of Object.keys(getters.ourDirectMessages)) {
+      // NOTE: skip DMs whose chatroom contracts are not synced yet
+      if (!state[chatRoomId] || !state[chatRoomId].users?.[getters.ourUsername]) {
+        continue
+      }
+      // NOTE: get only visible DMs for the current group
+      if (getters.ourDirectMessages[chatRoomId].groupContractID === state.currentGroupId &&
+        getters.ourDirectMessages[chatRoomId].visible &&
+        state[chatRoomId]
+      ) {
+        const users = Object.keys(state[chatRoomId].users)
+        const partners = users
+          .filter(username => username !== getters.ourUsername)
+          .sort((p1, p2) => {
+            const p1JoinedDate = new Date(state[chatRoomId].users[p1].joinedDate).getTime()
+            const p2JoinedDate = new Date(state[chatRoomId].users[p2].joinedDate).getTime()
+            return p1JoinedDate - p2JoinedDate
+          })
+        const lastJoinedPartner = partners[partners.length - 1]
```

Comment:
`lastJoinedParter` is chatroom member who has joined the chatroom for the last time.
His profile picture can be used as the picture of the direct message with the badge of the number of partners.